### PR TITLE
Added missing fields to the Thread Metadata 

### LINKED ISSUE #1092: Add missing fields to the Thread Metadata
Missing fields:
* `invitable`
* `create_timestamp`

Furthermore the meta data fields should be moved to a new ThreadMetadata object
https://discord.com/developers/docs/resources/channel#thread-metadata-object-thread-metadata-structure

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
closes #1092

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/entity/channel/thread/ThreadMetadata.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Whether this thread is locked.
+     * When a thread is locked, only users with MANAGE_THREADS can unarchive it.
+     *
+     * @return Whether this thread is locked.
+     */
+    boolean isLocked();
+
+
```
Comment (AhmedA1559) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/thread/ThreadMetadata.java:30]: Extraneous empty line

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java [hunk 2]
```diff
@@ -37,29 +37,12 @@ public interface ServerThreadChannel extends ServerChannel, TextChannel, Mention
      */
     int getMemberCount();
 
-
     /**
-     * Gets the duration for newly created threads, in minutes, to automatically
-     * archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080.
+     * Extra data about this thread.
```
Comment (AhmedA1559) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ServerThreadChannel.java:41]:
```suggestion
     * Gets the extra data about this thread.
```

**Review by AhmedA1559 — APPROVED** [review]
LGTM other than that.

**Comment by RealYusufIsmail** [comment]
done

**Comment by KILLEliteMaste** [comment]
So far it look good to me, but we should wait for #1089 to be merged first due to the field updates. Then this PR also needs a little rework to include updates for the new fields

**Pull request merged**